include/mbedtls/ctr_drbg.h

```diff
@@ -45,7 +48,13 @@
 #define MBEDTLS_ERR_CTR_DRBG_FILE_IO_ERROR                -0x003A  /**< Read or write error in file. */
 
 #define MBEDTLS_CTR_DRBG_BLOCKSIZE          16 /**< The block size used by the cipher. */
+
+#if !defined(MBEDTLS_CTR_DRBG_USE_128_BIT_KEY) /* Use default 256-bit key for CTR-DRBG. */
 #define MBEDTLS_CTR_DRBG_KEYSIZE            32 /**< The key size used by the cipher. */
+#else
+#pragma message("Warning: using smaller (128-bit) key size for CTR DRBG may reduce the security of some operations.")
```

Comment:
At a glance it seems that `#pragma message` is supported by reasonably modern versions of GCC, Clang, Visual and IAR. Those are the compilers that we absolutely must support. Have you checked whether it's supported by the specific versions we need to support?

I'd still prefer to use the more common `#warning`, and wrap it with `#if defined(MBEDTLS_DEPRECATED_WARNING)`. Since we recommend against using this feature, it is functionally deprecated.

Reply:
Hi @gilles-peskine-arm, 
I did a little digging in regards to #pragma #warning
1. I found that GCC versions equal or greater than 4.4.7 (version 4.3.6 doesn't) of GCC support #pragma message, (https://gcc.gnu.org/onlinedocs/gcc-4.4.7/gcc/Diagnostic-Pragmas.html#Diagnostic-Pragmas)
2. ancedotal search evidence indicates this is supported in visual studio 2010.
3. for clang it should be part of version 2.9 (https://github.com/llvm-mirror/clang/blob/release_29/test/Lexer/pragma-message.c)
4. for IAR version 5.4 supports it (couldn't find docs for earlier versions)(http://supp.iar.com/FilesPublic/UPDINFO/004916/arm/doc/EWARM_DevelopmentGuide.ENU.pdf)

If this level of support is not sufficient I can change the messages back to #warning (with the downsides noted above).